chips/rp2350/src/chip.rs

```diff
@@ -0,0 +1,174 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OxidOS Automotive 2025.
+
+//! Chip trait setup.
+
+use core::fmt::Write;
+use kernel::platform::chip::Chip;
+use kernel::platform::chip::InterruptService;
+
+use crate::clocks::Clocks;
+use crate::gpio::{RPPins, SIO};
+use crate::interrupts;
+use crate::resets::Resets;
+use crate::ticks::Ticks;
+use crate::timer::RPTimer;
+use crate::uart::Uart;
+use crate::xosc::Xosc;
+use cortexm33::{interrupt_mask, CortexM33, CortexMVariant};
+
+#[repr(u8)]
+pub enum Processor {
+    Processor0 = 0,
+    Processor1 = 1,
+}
+
+pub struct Rp2350<'a, I: InterruptService + 'a> {
+    mpu: cortexm33::mpu::MPU,
+    userspace_kernel_boundary: cortexm33::syscall::SysCall,
+    interrupt_service: &'a I,
+    sio: &'a SIO,
+    processor0_interrupt_mask: (u128, u128),
+    processor1_interrupt_mask: (u128, u128),
+}
+
+impl<'a, I: InterruptService> Rp2350<'a, I> {
+    pub unsafe fn new(interrupt_service: &'a I, sio: &'a SIO) -> Self {
+        Self {
+            mpu: cortexm33::mpu::MPU::new_ns(),
+            userspace_kernel_boundary: cortexm33::syscall::SysCall::new(),
+            interrupt_service,
+            sio,
+            processor0_interrupt_mask: interrupt_mask!(interrupts::PROC0_IRQ_CTI),
+            processor1_interrupt_mask: interrupt_mask!(interrupts::PROC1_IRQ_CTI),
```

Comment:
These now match the 0 -> 0. Is this something different about this chip compared to rp2340, or a difference in this implementation compared to the rp2340?

Reply:
Not really. I am thinking about making a separate crate named `rp2_common` where we would put all the duplicate code from `rp2040` and `rp2350`. There are some subtle differences between the chips, like the register addresses, but a considerable part of the code can be used by both of the chips.
This would follow how `pico-sdk` is structured.

Reply:
I'm just confused why these are different:

https://github.com/tock/tock/blob/422fcf6065e052e0d58d372b7f539486037c04f9/chips/rp2040/src/chip.rs#L51-L52

In the rp2040 proc0_mask is PROC1_IRQ and in the rp2350 proc0_mask is PROC0_IRQ. I think a comment clarifying the differences, maybe with a reference to the datasheet, would be helpful for future readers of the code.

Reply:
This was a mistake on my part, I'll modify it. It's a little strange that everything worked.